src/source/Ice/Network.c

```diff
@@ -133,12 +139,15 @@ STATUS createSocket(PKvsIpAddress pHostIpAddress, PKvsIpAddress pPeerAddress, KV
     pHostIpAddress->port = (UINT16) pHostIpAddress->family == KVS_IP_FAMILY_TYPE_IPV4 ? ipv4Addr.sin_port : ipv6Addr.sin6_port;
     *pSockFd = (INT32) sockfd;
 
-    // done at this point for UDP
-    CHK(protocol == KVS_SOCKET_PROTOCOL_TCP, retStatus);
-
-    if (connect(sockfd, peerSockAddr, addrLen) < 0) {
+    if (protocol == KVS_SOCKET_PROTOCOL_TCP && connect(sockfd, peerSockAddr, addrLen) < 0) {
         DLOGW("connect() failed with errno %s", strerror(errno));
-        CHK(FALSE, STATUS_SOCKET_CONNECT_FAILED);
+        CHK(FALSE, STATUS_SOCKET_CONNECT_FAILED);    
+    }
+
+    flags = fcntl(sockfd, F_GETFL, 0);
+    if (flags < 0 || fcntl(sockfd, F_SETFL, flags | O_NONBLOCK) < 0) {
+        DLOGW("failed to set socket flag with errno %s", strerror(errno));
+        CHK(FALSE, STATUS_SET_SOCKET_FLAG_FAILED);
```

Comment:
Doesn't let me comment in the CleanUp clause but seem that we might want to release the socket in case of an error